plugins/check_mysql.c

```diff
@@ -275,11 +277,29 @@ main (int argc, char **argv)
 			/* Save slave status in slaveresult */
 			snprintf (slaveresult, SLAVERESULTSIZE, "Slave IO: %s Slave SQL: %s Seconds Behind Master: %s", row[slave_io_field], row[slave_sql_field], seconds_behind_field!=-1?row[seconds_behind_field]:"Unknown");
 
-			/* Raise critical error if SQL THREAD or IO THREAD are stopped */
+			/* Raise critical error if SQL THREAD or IO THREAD are stopped, but only if there are no mysqldump threads running */
 			if (strcmp (row[slave_io_field], "Yes") != 0 || strcmp (row[slave_sql_field], "Yes") != 0) {
-				mysql_free_result (res);
-				mysql_close (&mysql);
-				die (STATE_CRITICAL, "%s\n", slaveresult);
+				MYSQL_RES *res_mysqldump;
+				MYSQL_ROW row_mysqldump;
+				unsigned int mysqldump_threads = 0;
+
+				if (mysql_query (&mysql, MYSQLDUMP_THREADS_QUERY) == 0) {
+					/* store the result */
+					if ( (res_mysqldump = mysql_store_result (&mysql)) != NULL) {
+						if (mysql_num_rows(res_mysqldump) == 1) {
+							if ( (row_mysqldump = mysql_fetch_row (res_mysqldump)) != NULL) {
+								mysqldump_threads = atoi(row_mysqldump[0]);
+							}
+						}
+						/* free the result */
+						mysql_free_result (res_mysqldump);
+					}
+				}
+				if (mysqldump_threads == 0) {
+					die (STATE_CRITICAL, "%s\n", slaveresult);
+				} else {
+					strlcat (slaveresult, " Mysqldump: in progress", SLAVERESULTSIZE);
```

Comment:
Is this BSD specific? If so, could you use `strncat`?

Reply:
Not BSD specific, will change it.

Reply:
Requested changes were addressed in https://github.com/monitoring-plugins/monitoring-plugins/pull/1718/commits/d8bc51f465e984001ccaca616e106da6b92fb015 .